--- mod_auth_mysql.c	2011-06-23 07:09:25.174015466 +0200
+++ mod_auth_mysql.c.oden	2011-06-23 07:09:44.426015456 +0200
@@ -810,12 +810,9 @@ static char hex2chr(char * in) {
 /* Checks scrambled passwords */
 static short pw_scrambled(POOL * pool, const char * real_pw, const char * sent_pw, const char * salt) {
   char * encrypted_sent_pw = PCALLOC(pool, SCRAMBLED_PASSWORD_CHAR_LENGTH+1);
-#ifdef SCRAMBLED_PASSWORD_CHAR_LENGTH_323   /* If we may need to use old password */
-  if (real_pw[0] != '*')
-    make_scrambled_password_323(encrypted_sent_pw, sent_pw);
-  else
-#endif
-    make_scrambled_password(encrypted_sent_pw, sent_pw);
+  void my_make_scrambled_password(char *to, const char *password,             
+                                  size_t pass_len);
+  my_make_scrambled_password(encrypted_sent_pw, sent_pw, strlen(sent_pw));
   return strcmp(real_pw, encrypted_sent_pw) == 0;
 }
 
